frontend/src/components/RegisterForm.vue

```diff
@@ -1,60 +1,160 @@
 <script lang="ts" setup>
 import MultiStepForm from '@/components/MultiStepForm.vue'
 import { Api } from '@/utils/api'
-import { ref } from 'vue'
-import Alert from './Alert.vue'
-
-let formData = {
-  username: '',
-  email: '',
-  password: '',
-  age: 0,
-  first_name: '',
-  last_name: '',
-  gender: '' as 'M' | 'F' | 'O',
-  sexual_preferences: '' as 'M' | 'F' | 'A' | 'O',
-  biography: '',
-}
+import { ref, computed, onMounted, nextTick } from 'vue'
+import FeedbackToast from '@/components/FeedbackToast.vue'
+
+const formData = {
+  username: null,
+  email: null,
+  password: null,
+  birthday: null,
+  first_name: null,
+  last_name: null,
+  gender: null as 'M' | 'F' | 'O',
+  sexual_preferences: null as 'M' | 'F' | 'A' | 'O',
+  biography: null,
+};
+
+const refs = ref({})
+const step = ref()
+const currentStep = computed(() => {
+  return step?.value?.currentStep || 0;
+});
 
 const totalSteps = 4
-const step = ref(0)
-const hasError = ref(false)
-const titleAlert = ref('test')
+const feedbackRef = ref()
+const maxAge = ref()
+
+
+function setMaxAge()
+{
+  const maxDate = new Date();
+  maxDate.setFullYear(new Date().getFullYear() - 18);
+
+  const formattedDate = maxDate.toISOString().split('T')[0];
+  maxAge.value = formattedDate
+}
+setMaxAge()
 
 async function handleSubmit() {
+  console.log(formData)
   try {
     const req = await Api.post('/auth/register').send(formData)
     const data = await req.json()
 
     if (req.status == 400) {
-      hasError.value = true
-      titleAlert.value = data.message
+      feedbackRef.value.addError(data.message)
+      focusInput(getErrorCode(data.message))
     } else {
       Api.post('email/verif').send({ email: formData.email })
+      step.value.nextStep()
     }
   } catch (error) {
     console.error('Error during API request:', error)
-    hasError.value = true
-    titleAlert.value = 'An unexpected error occurred.'
+    feedbackRef.value.addError('An unexpected error occurred.')
   }
 }
 
-function handleChangeStep(n: number) {
-  step.value = n
+function getErrorCode(message)
+{
+  if (message.search("username") != -1)
+    return 0
+  else if (message.search("email") != -1)
+    return 1
+  else if (message.search("password") != -1)
+    return 2
+  else if (message.search("first_name") != -1)
+    return 3
+  else if (message.search("last_name") != -1)
+    return 4
+  else if (message.search("birthday") != -1)
+    return 5
+  else if (message.search("gender") != -1)
+    return 6
+  else if (message.search("sexual_preferences") != -1)
+    return 7
+  else if (message.search("biography") != -1)
+    return 8
```

Comment:
Isn't there a way to do something cleaner?

Reply:
It's my question that ! XD

Reply:
ensure that it is not the validators that return the error, or that the error or return code is customizable?

Reply:
I don't know.
I suggest it's a problem we put off until later.

Reply:
Yes, it's not a problem, it's just for the aspect of the code